// fable-import-react/Fable.Helpers.React.js
(function (global, factory) {
    if (typeof define === "function" && define.amd) {
        define(["exports", "babel-runtime/core-js/object/get-own-property-names", "fable-core", "react"], factory);
    } else if (typeof exports !== "undefined") {
        factory(exports, require("babel-runtime/core-js/object/get-own-property-names"), require("fable-core"), require("react"));
    } else {
        var mod = {
            exports: {}
        };
        factory(mod.exports, global.getOwnPropertyNames, global.fableCore, global.react);
        global.unknown = mod.exports;
    }
})(this, function (exports, _getOwnPropertyNames, _fableCore, _react) {
    "use strict";

    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    exports.toPlainJsObj = exports.Component = exports.Props = undefined;

    var _getOwnPropertyNames2 = _interopRequireDefault(_getOwnPropertyNames);

    function _interopRequireDefault(obj) {
        return obj && obj.__esModule ? obj : {
            default: obj
        };
    }

    var Props = exports.Props = function ($exports) {
        var CSSProp = $exports.CSSProp = class CSSProp {
            constructor() {
                this.Case = arguments[0];
                this.Fields = [];

                for (var i = 1; i < arguments.length; i++) {
                    this.Fields[i - 1] = arguments[i];
                }
            }

        };

        _fableCore.Util.setInterfaces(CSSProp.prototype, ["Fable.Helpers.React.Props.ICSSProp"]);

        var Prop = $exports.Prop = class Prop {
            constructor() {
                this.Case = arguments[0];
                this.Fields = [];

                for (var i = 1; i < arguments.length; i++) {
                    this.Fields[i - 1] = arguments[i];
                }
            }

        };

        _fableCore.Util.setInterfaces(Prop.prototype, ["Fable.Helpers.React.Props.IHTMLProp"]);

        var DOMAttr = $exports.DOMAttr = class DOMAttr {
            constructor() {
                this.Case = arguments[0];
                this.Fields = [];

                for (var i = 1; i < arguments.length; i++) {
                    this.Fields[i - 1] = arguments[i];
                }
            }

        };

        _fableCore.Util.setInterfaces(DOMAttr.prototype, ["Fable.Helpers.React.Props.IHTMLProp"]);

        var HTMLAttr = $exports.HTMLAttr = class HTMLAttr {
            constructor() {
                this.Case = arguments[0];
                this.Fields = [];

                for (var i = 1; i < arguments.length; i++) {
                    this.Fields[i - 1] = arguments[i];
                }
            }

        };

        _fableCore.Util.setInterfaces(HTMLAttr.prototype, ["Fable.Helpers.React.Props.IHTMLProp"]);

        var SVGAttr = $exports.SVGAttr = class SVGAttr {
            constructor() {
                this.Case = arguments[0];
                this.Fields = [];

                for (var i = 1; i < arguments.length; i++) {
                    this.Fields[i - 1] = arguments[i];
                }
            }

        };

        _fableCore.Util.setInterfaces(SVGAttr.prototype, ["Fable.Helpers.React.Props.IProp"]);

        return $exports;
    }({});

    var Component = exports.Component = class Component extends _react.Component {
        constructor(props, state) {
            super(props);
            this.state = state;
        }

    };

    var toPlainJsObj = exports.toPlainJsObj = function (o) {
        return _fableCore.Seq.fold((o2, k) => {
            return o2[k] = o[k], o2;
        }, {}, (0, _getOwnPropertyNames2.default)(o));
    };
});
